chainer/links/loss/hierarchical_softmax.py

```diff
@@ -82,15 +103,33 @@ def __init__(self, tree):
         parser.parse(tree)
         paths = parser.get_paths()
         codes = parser.get_codes()
+        parent2child = parser.get_parent2child()
         n_vocab = max(paths.keys()) + 1
-
+        self.n_vocab = n_vocab
         self.paths = numpy.concatenate(
-            [paths[i] for i in range(n_vocab) if i in paths])
+            [paths[i] for i in six.moves.range(n_vocab) if i in paths])
         self.codes = numpy.concatenate(
-            [codes[i] for i in range(n_vocab) if i in codes])
+            [codes[i] for i in six.moves.range(n_vocab) if i in codes])
+
+        def _convert(x):
```

Comment:
Thank you for your advice.
I notice that the size of`self.parent2child ` can be `(n_vocab, 2)`.